### ISSUE TITLE
Can you tell me the reason for randomness?

### ISSUE THREAD

**Issue by noparkee** [uber-research/LaneGCN#30]
Hello, Thanks for your nice project.

I tried to train the model several times without code editing, then I found that the difference in performance of each trial was quite large.
There is no big difference when testing multiple times with one checkpoint, so randomness seems to occur during the learning process.
Do you have any idea what could be the reason?

Thank you!

**Comment by FUTUREEEEEE** [comment]
i think the randomness is caused by the usage of nondeterministic-algorithms , which means even you set all the random seed ,the result will still have some randomness.  check it out by add `torch.use_deterministic_algorithms(True)` into the main function .

**Comment by FUTUREEEEEE** [comment]
> Hello, Thanks for your nice project.
> 
> I tried to train the model several times without code editing, then I found that the difference in performance of each trial was quite large. There is no big difference when testing multiple times with one checkpoint, so randomness seems to occur during the learning process. Do you have any idea what could be the reason?
> 
> Thank you!

btw,have you try to submit the test result to eval AI ? how is the outcome? i try to submit it result using the pretrained model provided by the author and my own training result , it turns out very bad....

**Comment by noparkee** [comment]
Thank you for you answer. I'll check it later.
I didn't submit the evaluation, just checked the validation result. I am sorry for not being helpful 😅

**Comment by huanmanlou** [comment]
> > Hello, Thanks for your nice project.
> > I tried to train the model several times without code editing, then I found that the difference in performance of each trial was quite large. There is no big difference when testing multiple times with one checkpoint, so randomness seems to occur during the learning process. Do you have any idea what could be the reason?
> > Thank you!
> 
> btw,have you try to submit the test result to eval AI ? how is the outcome? i try to submit it result using the pretrained model provided by the author and my own training result , it turns out very bad....

I recently obtained the h5 file using the pretrained model provided by the author, but the results are much worse than what the author described in the paper. Do you know the reason? Thank you
